eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -770,18 +771,39 @@ debugPlutus version db =
                   PlutusData d = pData
                in case sl of
                     SPlutusV1 ->
-                      case PV1.evaluateScriptRestricting pv v cm eu script d of
+                      case deserialiseAndEvaluateScript PlutusV1 pv v cm eu script d of
                         (logs, Left e) -> DebugInfo logs (PlutusErrorV1 e) pd
                         (_, Right ex) -> DebugSuccess ex
                     SPlutusV2 ->
-                      case PV2.evaluateScriptRestricting pv v cm eu script d of
+                      case deserialiseAndEvaluateScript PlutusV2 pv v cm eu script d of
                         (logs, Left e) -> DebugInfo logs (PlutusErrorV2 e) pd
                         (_, Right ex) -> DebugSuccess ex
                     SPlutusV3 ->
-                      case PV3.evaluateScriptRestricting pv v cm eu script d of
+                      case deserialiseAndEvaluateScript PlutusV3 pv v cm eu script d of
                         (logs, Left e) -> DebugInfo logs (PlutusErrorV3 e) pd
                         (_, Right ex) -> DebugSuccess ex
 
+-- TODO: this function is supposed to exist temporarily to cope with the breaking changes
+-- in https://github.com/input-output-hk/plutus/pull/5538, until script deserialisation
+-- is properly fixed in the ledger.
```

Comment:
Can we be more specific? Do we have a JIRA ticket for this? I think the specific issue is "until the ledger changes to deserialize scripts as part of transaction deserialisation", right?

Reply:
We have a related ticket in ledger: https://github.com/input-output-hk/cardano-ledger/issues/3717

I think it is enough for the context.

> until the ledger changes to deserialize scripts as part of transaction deserialisation

Just a side comment. We'll not be able to make plutus script deserialization as part of transaction deserialization, because plutus script deserialization is allowed to change for intra-era hardforks, which is not supported by the versioned serialization interface used by ledger.

However, this doesn't mean we can't avoid redundant deserialization in the rules, so this work will provide a nice optimization regardless.